SourceCode/raytracer/Intersection.h

```diff
@@ -0,0 +1,21 @@
+#pragma once
+#include <string>
+
+enum class Intersection {
```

Comment:
Usually an IntersectioData structure is used, which holds the hit point, hit normal, ray t, ... Try to incorporate this idea in your code.